cpp/src/io/json/write_json.cu

```diff
@@ -254,38 +182,235 @@ std::unique_ptr<column> struct_to_strings(table_view const& strings_columns,
+        // column_name: value
+        d_strviews[this_index]     = str_pair(d_col_name.data(), d_col_name.size_bytes());
+        d_strviews[this_index + 1] = str_pair(d_str.data(), d_str.size_bytes());
+      }
+      // suffix
+      if (col == tbl.num_columns() - 1) {
+        d_strviews[this_index + 2] = str_pair(row_suffix.data(), row_suffix.size_bytes());
+      }
+    });
+  auto joined_col = make_strings_column(d_strviews, stream, mr);
+
+  // gather from offset and create a new string column
+  rmm::device_uvector<size_type> d_strview_offsets(strings_columns.num_rows() + 1, stream);
+  thrust::sequence(rmm::exec_policy(stream),
+                   d_strview_offsets.begin(),
+                   d_strview_offsets.end(),
+                   0,
+                   num_str_views_per_row);
+  auto old_offsets = strings_column_view(joined_col->view()).offsets();
+  auto row_string_offsets =
+    std::move(cudf::detail::gather(cudf::table_view{{old_offsets}},
+                                   d_strview_offsets,
+                                   cudf::out_of_bounds_policy::DONT_CHECK,
+                                   cudf::detail::negative_index_policy::NOT_ALLOWED,
+                                   stream,
+                                   mr)
+                ->release()
+                .front());
+  return make_strings_column(
+    strings_columns.num_rows(),
+    std::move(row_string_offsets),
+    std::move(joined_col->release().children[strings_column_view::chars_column_index]),
+    0,
+    {});
+}
+
+/**
+ * @brief Concatenates a list of strings columns into a single strings column.
+ *
+ * @param lists_strings Column containing lists of strings to concatenate.
+ * @param list_prefix String to place before each list. (typically [)
+ * @param list_suffix String to place after each list. (typically ])
+ * @param element_separator String that should inserted between strings of each list row.
+ * @param element_narep String that should be used in place of any null strings.
+ * @param stream CUDA stream used for device memory operations and kernel launches.
+ * @param mr Device memory resource used to allocate the returned column's device memory.
+ * @return New strings column with concatenated results.
+ */
+std::unique_ptr<column> join_list_of_strings(lists_column_view const& lists_strings,
+                                             string_view const list_prefix,
+                                             string_view const list_suffix,
+                                             string_view const element_separator,
+                                             string_view const element_narep,
+                                             //  string_view const list_narep,
+                                             rmm::cuda_stream_view stream,
+                                             rmm::mr::device_memory_resource* mr)
+{
```

Comment:
Algorithm used:
-  calculate #strviews per list using null mask, and list_offsets.
-  scan #strviews to get #strviews(scanned)
-  create label segments.
-  sublist_index= index - offsets[label]
-  #strviews(scanned)[label] + sublist_index = string_view index +1, +2
-  use above 2 to scatter element, element_seperator
-  scatter list_prefix, list_suffix to the right place using list_offsets
-  make_strings_column() and gather offsets, based on #strviews(scanned).

Reply:
Perhaps add this as a comment in the code?

Reply:
It was in the code earlier. I removed it. I will add it.